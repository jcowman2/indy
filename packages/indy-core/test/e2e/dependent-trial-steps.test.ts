import { Dependent, IndyError, Runner } from "../..";

let testClient: Dependent;

const newRunner = () => {
    return new Runner()
        .on("info", data => console.log(data.message))
        .on("debug", data => console.log(data.message))
        .on("error", data => console.log(data.message));
};

beforeAll(async done => {
    const indy = newRunner();
    testClient = await indy.load("indy-test-client", {
        path: "./demo/indy-test-client",
        initCommands: ["npm install"],
        buildCommands: [],
        testCommands: ["npm test"]
    });
    done();
});

afterAll(async done => {
    await testClient.swapDependency(
        "@jcowman/indy-broken-lib",
        "@jcowman/indy-broken-lib"
    );
    done();
});

test("e2e: Dependent trial steps", async done => {
    await testClient.init();
    await testClient.build();

    expect.assertions(2);

    // Verify current build is failing.
    try {
        await testClient.test();
    } catch (e) {
        expect(e instanceof IndyError).toBeTruthy();
        expect(e.message).toEqual(
            "One or more verification tests failed. (Code: 403) Cause: An error occurred during 'npm test'. See output for more information. (Code: 411)"
        );
    }

    await testClient.swapDependency(
        "@jcowman/indy-broken-lib",
        "../indy-fixed-lib" // Relative to where we're running the CLI from: indy-broken-lib
    );

    await testClient.test(); // Should run successfully now.

    done();
});
