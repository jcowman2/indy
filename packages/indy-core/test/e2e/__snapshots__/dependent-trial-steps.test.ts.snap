// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e: Dependent trial steps 1`] = `
"Initializing @jcowman/indy-test-client...
Spawning command 'npm' with args [install] where cwd = [PATH REMOVED]npm WARN @jcowman/indy-test-client@1.0.0 No description
npm WARN @jcowman/indy-test-client@1.0.0 No repository field.

<Internal Process STDOUT>
<Internal Process STDOUT>
Completed command 'npm install' with exit code 0.
@jcowman/indy-test-client initalized successfully.
Building @jcowman/indy-test-client...
@jcowman/indy-test-client built successfully.
Running @jcowman/indy-test-client's verification tests...
Spawning command 'npm' with args [test] where cwd = [PATH REMOVED]<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
npm ERR! Test failed.  See above for more details.
Command 'npm test' failed with exit code 1. Bail = true.
An error occurred during 'npm test'. See output for more information.
One or more verification tests failed.
Swapping @jcowman/indy-test-client's @jcowman/indy-broken-lib dependency...
Spawning command 'npm' with args [uninstall,@jcowman/indy-broken-lib] where cwd = [PATH REMOVED]npm WARN @jcowman/indy-test-client@1.0.0 No description
npm WARN @jcowman/indy-test-client@1.0.0 No repository field.

<Internal Process STDOUT>
<Internal Process STDOUT>
Completed command 'npm uninstall @jcowman/indy-broken-lib' with exit code 0.
Spawning command 'npm' with args [install,../indy-fixed-lib] where cwd = [PATH REMOVED]npm WARN @jcowman/indy-test-client@1.0.0 No description
npm WARN @jcowman/indy-test-client@1.0.0 No repository field.

<Internal Process STDOUT>
<Internal Process STDOUT>
Completed command 'npm install ../indy-fixed-lib' with exit code 0.
@jcowman/indy-test-client's @jcowman/indy-broken-lib dependency swapped from ^1.0.0 to ^1.0.0 successfully.
Running @jcowman/indy-test-client's verification tests...
Spawning command 'npm' with args [test] where cwd = [PATH REMOVED]<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
<Internal Process STDOUT>
Completed command 'npm test' with exit code 0.
"
`;
